Automattic/wp-calypso
# CPT: Set list screen title using post type label

Related: #3796
Supercedes: #4396

This pull request seeks to:
- Rename `state.page` to `state.documentHead` (and general refactoring for action creator name consistency)
- Rename `<Page />` to `<DocumentHead />` (and minor refactoring)
- Account for changes in site when applying title, by refactoring `lib/screen-title/index` Redux hooks to setting title in response to dispatch
- Update title on CPT list screens using post type's label

__Testing instructions:__

Ensure that the title is correctly set when navigating to a custom post type list screen, that the site suffix is appended, and that the site suffix changes when switching between sites. Additionally ensure that after navigating away from the CPT list page, titles continue to update correctly.

1. Navigate to [custom post type list screen for posts](http://calypso.localhost:3000/types/post)
2. Select a site
3. Note that the page title is "Posts" and includes suffix for selected site
4. Switch sites
5. Note that selected site suffix changes in title
6. Continue navigating around Calypso and optionally switching sites
7. Note that titles update correctly
